### ISSUE TITLE
Permissions error when installing on Windows 7 machine

### ISSUE THREAD

**Issue by HekTron802** [gjtorikian/commonmarker#61]
I'm a complete Noob, when it comes to this so please let me know if I can provide any additional information.

When I try to install commonmarker I get the following error:
```
C:\Users\User\Documents\GitHub\MarlinDocumentation>gem install commonmarker -v '0.17.7'
Temporarily enhancing PATH for MSYS/MINGW...
ERROR:  While executing gem ... (Errno::EACCES)
    Permission denied @ rb_file_s_symlink - (libcmark-gfmextensions.0.28.3.gfm.11.dylib, C:/Ruby24-x64/lib/ruby/gems/2.4.0/gems/commonmarker-0.17.7/ext/commonmarker/cmark-upstream/build/extensions/lib
cmark-gfmextensions.dylib)
```

I tried playing around with the Windows permissions but that didn't work.

Please help.

**Comment by kivikakk** [comment]
Thanks so much for your report! I'm looking into this now; it seems to be a Windows-specific issue in our build process, and definitely not your fault!

**Comment by kivikakk** [comment]
Okay, this is annoying but I totally cannot reproduce this under any circumstance:

```
~\Documents\GitHub\commonmarker [master ≡ +1 ~0 -0 !]> gem install commonmarker
Fetching: commonmarker-0.17.7.gem (100%)
Temporarily enhancing PATH to include DevKit...
Building native extensions.  This could take a while...
Successfully installed commonmarker-0.17.7
```

Continuing to investigate.

**Comment by kivikakk** [comment]
Gem version 0.17.7.1 has been released which should address this issue; please let me know how it goes for you!

**Comment by HekTron802** [comment]
Thank you!

I did have an error when I tried to install commonmarker this way:

```
C:\Users\User\Documents\GitHub\MarlinDocumentation>gem install commonmarker -v '0.17.7'
Temporarily enhancing PATH for MSYS/MINGW...
Building native extensions. This could take a while...
Successfully installed commonmarker-0.17.7
Parsing documentation for commonmarker-0.17.7
Installing ri documentation for commonmarker-0.17.7
ERROR:  While executing gem ... (ArgumentError)
    invalid byte sequence in UTF-8
```
However when I used

`C:\Users\User\Documents\GitHub\MarlinDocumentation>bundle install --path vendor/bundle`

Commonmaker and a bunch of other stuff I needed installed fine.

I'm going to leave this ticket open incase that you want to use it for the invalid byte issue, but feel free to close it.

Thanks again for your help with this!

**Comment by cmegown** [comment]
I can confirm that 0.17.7.1 resolved this issue for me.

**Comment by kivikakk** [comment]
@HekTron802 thank you for the report! I'll look into the `ri` generation bug, but I'm glad it's otherwise working for you. I'd hit that once before, and just repeating the install attempt the same way seemed to work also.

**Comment by kivikakk** [comment]
I think the UTF-8 issue should be fixed as of 0.17.7.1, but if not, please let me know.
